compiler/lib/src/main/scala/util/Error.scala

```diff
@@ -195,8 +195,12 @@ sealed trait Error {
       case SemanticError.MissingConnection(loc, matchingLoc) =>
         Error.print (Some(loc)) ("no match for this connection")
         printMatchingLoc(matchingLoc)
+      case SemanticError.MissingPortMatching(loc) =>
+        Error.print (Some(loc)) ("unmatched connection must go from or to a matched port")
       case SemanticError.MissingPort(loc, specMsg, portMsg) =>
         Error.print (Some(loc)) (s"component with $specMsg must have $portMsg")
+      case SemanticError.MultipleConnectionsAtPortIndex(portIndexString) =>
+        Error.print (None) (s"multiple connections appear at port index $portIndexString")
```

Comment:
Re-word error, something along the lines of "duplicate connection at matched port"